torch_xla/csrc/tensor_util.cpp

```diff
@@ -869,7 +897,19 @@ std::vector<torch::lazy::BackendDataPtr> CreateTensorsData(
     const std::vector<std::string>& devices, bool transfer_async) {
   TORCH_LAZY_TIMED("TensorToData");
   XLA_CHECK_EQ(tensors.size(), devices.size());
-  if (transfer_async) {
+  if (ShardingUtil::UseVirtualDevice()) {
+    // If running in SPMD mode, replicate the data across addressable devices
+    std::vector<std::string> local_devices =
+        xla::ComputationClient::Get()->GetLocalDevices();
+    std::vector<xla::ComputationClient::DataPtr> handles;
+    for (size_t i = 0; i < tensors.size(); ++i) {
+      auto device = ParseDeviceString(devices[i]);
+      auto shape = CreateComputationShapeFromTensor(tensors[i], &device);
+      handles.push_back(TransferToServerReplicated(tensors[i], devices[i],
```

Comment:
It looks like we unconditionally replicate all the data? I thought tiled data shouldn't be replicated?

Reply:
Sounds good, I'll add a comment. The `else if` should still be necessary for some non-SPMD scenarios, I'm just making it so that the SPMD case takes precedence here.

Reply:
Yea, it's more like a NIT. To me, it feels like the logic is:
1. SPMD
2. Non SPMD:
2.1. transfer_async
2.2. transfer_sync

But then your coding is:
1. SPMD
2. transfer_async
3. transfer_sync

Reply:
Ah gotcha! That makes sense. So something like

```
if (SPMD) {
...
return;
}

if (transfer_async) {
...
} else {
...
}
```

I'll update this part to make the logic more clear.

Reply:
ehh, I will get rid of `transfer_async` once we get rid of XRT, that was a hacked I did which make codes really messy.